spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchQueryScan.java

```diff
@@ -63,7 +64,7 @@ class SparkBatchQueryScan extends SparkScan implements SupportsRuntimeFiltering
 
   private static final Logger LOG = LoggerFactory.getLogger(SparkBatchQueryScan.class);
 
-  private final TableScan scan;
+  private final Scan<?, ?, ?> scan;
```

Comment:
I considered defining this variable type as below:
```
private final Scan<?, ? extends ScanTask, ? extends ScanTaskGroup<? extends ScanTask>> scan;
```

But it seemed like a very hard to read definition.

Reply:
it would be nice if Java had type macros  :(

Reply:
We could do something like this (if it is any better):

```
private final Scan<?, ? extends ScanTask, ?> scan
private final Scan<?, ? extends ScanTask, ? extends ScanTaskGroup<?>> scan
```

Reply:
The last seems reasonable. Unless there is a totally different idea.

Reply:
It's verbose, but it is a bit less scary to me than just ?,? ,?

Reply:
Changed.